python/plugins/processing/gui/ExtentSelectionPanel.py

```diff
@@ -205,10 +205,10 @@ def setValueFromRect(self, r):
         self.dialog.activateWindow()
 
     def getValue(self):
-        if unicode(self.leText.text()).strip() != '':
-            return unicode(self.leText.text())
-        else:
+        if unicode(self.leText.text()).strip() == 'auto':
             return self.getMinCoveringExtent()
+        else:
+            return unicode(self.leText.text())
```

Comment:
Is it required to cast this explicitly to unicode?
Skipping explicit unicode casts wherever possible will make code more portable between python 2/3.